File: 1/solution.ts
// Path: 2022/1/solution.ts

export function part1(input: string): number {
    const inputArray = input.split("\n");
    const elves = [];
    let total = 0;

    for (let i = 0; i < inputArray.length; i++) {
        if (inputArray.at(i) !== "") {
            total += parseInt(inputArray.at(i) ?? "0");
        } else {
            elves.push(total);
            total = 0;
        }
    }

    return Math.max(...elves);
}

export function part2(input: string): number {
    const inputArray = input.split("\n");
    const elves = [];
    let total = 0;

    for (let i = 0; i < inputArray.length; i++) {
        if (inputArray.at(i) !== "") {
            total += parseInt(inputArray.at(i) ?? "0");
        } else {
            elves.push(total);
            total = 0;
        }
    }

    elves.sort((a, b) => a - b);

    return (elves.at(-1) ?? 0) + (elves.at(-2) ?? 0) + (elves.at(-3) ?? 0);
}

const input = Deno.readTextFileSync("./1/input.txt");

console.log("Part 1:", part1(input));
console.log("Part 2:", part2(input));

File: 1/solution.py
# Path: 2022/1/solution.py


def part1(input: list[str]) -> int:
    elves = []

    total = 0
    for i in input:
        if i != "\n":
            total += int(i)
        else:
            elves.append(total)
            total = 0

    return max(elves)


def part2(input: list[str]) -> int:
    elves = []

    total = 0
    for i in input:
        if i != "\n":
            total += int(i)
        else:
            elves.append(total)
            total = 0

    elves.sort()
    return elves[-1] + elves[-2] + elves[-3]


if __name__ == "__main__":
    input = open("./1/input.txt", "r").readlines()
    print("Part 1: " + str(part1(input)))
    print("Part 2: " + str(part2(input)))
